### PR TITLE
Bug: Rename temp files only on success

### PR DESCRIPTION
**Pull request by nakulkar-msft**

### CODE REVIEW

**Review by mohsha-msft — APPROVED** [review]
LGTM

**Review by zezha-msft — APPROVED** [review]

File: ste/xfer-remoteToLocal-file.go [hunk 1]
```diff
@@ -352,11 +352,11 @@ func epilogueWithCleanupDownload(jptm IJobPartTransferMgr, dl downloader, active
...
-			if !strings.EqualFold(info.Destination, common.Dev_Null) {
-				err := os.Rename(info.getTempDownloadPath(), info.Destination)
-				if  err != nil {
+			//downloaded and not corrupt. Infact, post this point we should only log errors and
+			//not fail the transfer.
+			if err == nil && !strings.EqualFold(info.Destination, common.Dev_Null) {
+				renameErr := os.Rename(info.getTempDownloadPath(), info.Destination)
+				if  renameErr != nil {
 					jptm.LogError(info.Destination, fmt.Sprintf(
 								  "Failed to rename. File at %s", info.getTempDownloadPath()), err)
```
Comment (kayoshino) [inline ste/xfer-remoteToLocal-file.go:361]: @nakulkar-msft @mohsha-msft @zezha-msft  Should this line be referencing `renameErr` and not `err`? I'm hitting a "nil pointer dereference" error on this line when I call azcopy with --from-to=FileLocal.

I can open a separate issue if you'd like.

Comment (zezha-msft) [reply ste/xfer-remoteToLocal-file.go:361]: Thanks for reaching out!

`err` should be nil here inside the if statement on line 357. 

Please do open a new issue, thank you!

Comment (kayoshino) [reply ste/xfer-remoteToLocal-file.go:361]: No problem! Thanks for the quick response. Issue created.

https://github.com/Azure/azure-storage-azcopy/issues/1421

**Pull request merged**